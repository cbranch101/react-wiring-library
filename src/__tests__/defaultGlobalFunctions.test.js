import React, {useState} from 'react'
import {cleanup} from '@testing-library/react'
import buildWiring from '../index'

afterEach(cleanup)

const Input = () => {
  const [focused, setFocus] = useState(false)
  return (
    <input
      data-focused={focused || undefined}
      onFocus={() => setFocus(true)}
      onBlur={() => setFocus(false)}
      data-testid="input"
      type="text"
    />
  )
}

const blurFixture = <Input />

const blurWiring = {
  children: {
    input: {
      findValue: 'input',
      serialize: val => {
        const hasFocus = !!val.getAttribute('data-focused')
        return `[${hasFocus ? 'Focused: ' : ''}${val.value}]`
      },
    },
  },
}

describe('Default functions returned with all wiring', () => {
  test('focus, blur and typeInto, should behave as expected', async () => {
    const getRender = buildWiring(blurWiring)
    const render = getRender(['input'])
    const {findInput} = render(blurFixture)
    const {input, blur, focus, typeInto} = await findInput()
    expect(input).toMatchSnapshot('initial render')
    focus()
    expect(input).toMatchSnapshot('after calling focus')
    typeInto('Some stuff')
    expect(input).toMatchSnapshot('after typing Some stuff')
    blur()
    expect(input).toMatchSnapshot('after blurring')
  })
})
