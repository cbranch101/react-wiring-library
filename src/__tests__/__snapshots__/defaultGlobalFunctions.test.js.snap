// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default functions returned with all wiring focus, blur and typeInto, should behave as expected: after blurring 1`] = `[Some stuff]`;

exports[`Default functions returned with all wiring focus, blur and typeInto, should behave as expected: after calling focus 1`] = `[Focused: ]`;

exports[`Default functions returned with all wiring focus, blur and typeInto, should behave as expected: after typing Some stuff 1`] = `[Focused: Some stuff]`;

exports[`Default functions returned with all wiring focus, blur and typeInto, should behave as expected: initial render 1`] = `[]`;
